lib/ide/ast/impl/src/lib.rs

```diff
@@ -19,7 +19,7 @@ use uuid::Uuid;
 /// A sequence of AST nodes, typically the "token soup".
 pub type Stream<T> = Vec<T>;
 
-
+type IDMap = Vec<((usize,usize),ID)>;
```

Comment:
Please use a struct instead of nested tuples. We have banned using tuples almost everywhere as it is really hard to understand what they are. For example, what are these `usizes`?